### PR TITLE
decode bytes to str

### PR DESCRIPTION
**Pull request by bmp4070**
and deploy.py to upload to s3

### CODE REVIEW

File: deploy.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            's3', 'put-object', '--body', fpath, '--bucket', 'sfdc-ansible', '--key',
+            f'{folder}/' + fpath.split('/')[-1]
+        ])
+
+
+if __name__ == "__main__":
+    import docopt
+    opts = docopt.docopt(__doc__)
+    uploads = None
+    if uploads == 'all':
```
Comment (ClarenceGardner) [inline deploy.py]: This was just set to None

File: o4/o4_pyforce.py [hunk 2]
```diff
@@ -125,7 +126,10 @@ def to_str(r):
 
         def dec(a):
             if hasattr(a, 'decode'):
-                return a.decode(sys.stdout.encoding)
+                try:
+                    a = a.decode(sys.stdout.encoding)
+                except UnicodeDecodeError:
+                    a = a.decode(sys.stdout.encoding, errors='ignore')
```
Comment (ClarenceGardner) [inline o4/o4_pyforce.py]: Why not just do the ignoring unconditionally?

**Comment by ClarenceGardner** [comment]
Wouldn't deploy.py be better in blt/blt-code/bin (in git.soma)

File: o4/o4_pyforce.py [hunk 3]
```diff
@@ -125,7 +126,7 @@ def to_str(r):
 
         def dec(a):
             if hasattr(a, 'decode'):
-                return a.decode(sys.stdout.encoding)
+                a = a.decode(sys.stdout.encoding, errors='ignore')
```
Comment (philipbergen) [inline o4/o4_pyforce.py]: You should leave the return, reassigning to a is just slower and can have unknown side effects if a is not a regular `bytes`. (I know it is, but think about the pattern).

File: o4/o4_pyforce.py [hunk 4]
```diff
@@ -30,8 +30,9 @@ def __init__(self, *args):
         self.stderr = NamedTemporaryFile()
         if os.environ.get('DEBUG', ''):
             print(f'## p4', *self.args, file=sys.stderr)
-        self.pope = Popen(
-            ['p4', '-vnet.maxwait=60', '-G'] + self.args, stdout=PIPE, stderr=self.stderr)
+        self.pope = Popen(['p4', '-vnet.maxwait=60', '-G'] + self.args,
+                          stdout=PIPE,
+                          stderr=self.stderr)
```
Comment (philipbergen) [inline o4/o4_pyforce.py]: Is your yapf set up correctly and updated to the latest version? Looks like you're wrapping at 80

Comment (philipbergen) [reply o4/o4_pyforce.py]: Ah, yes it is. I get this wrapping with the latest version. Sigh.

File: o4/o4_pyforce.py [hunk 5]
```diff
@@ -125,7 +126,7 @@ def to_str(r):
 
         def dec(a):
             if hasattr(a, 'decode'):
-                return a.decode(sys.stdout.encoding)
+                return a.decode(sys.stdout.encoding, errors='ignore')
```
Comment (philipbergen) [inline o4/o4_pyforce.py]: Just pointing out that *this* is the only code change in this PR.

**Comment by amateurhuman** [comment]
@bmp4070 You've been invited to the org. Once you accept the invitation you can refresh the CLA check by going to https://cla.salesforce.com/status/salesforce/o4/pull/14

**Pull request merged**